stages/stage_parsing_entry.go

```diff
@@ -33,13 +37,14 @@ func (s *StageParsingEntry) Close() {
 }
 
 // NewStageParsingEntry is a StageParsingEntry constructor
-func NewStageParsingEntry(input <-chan *common.Entry, parsers []ParserFunction,
+func NewStageParsingEntry(input <-chan *common.Entry, parseDocker, parseContainerD ParserFunction,
 	parserDefault ParserFunctionDefault, logger logging.Logger) *StageParsingEntry {
 	stage := &StageParsingEntry{
 		stage: stage{wg: &sync.WaitGroup{}, logger: logger},
 
-		parsers:      parsers,
-		parseDefault: parserDefault,
+		parseDockerFormat:     parseDocker,
```

Comment:
The reason to have formats parsers passed though constructor are tests?
Because just calling functions in `switch` is just seem more simple and less code to me.

Reply:
Yes, mostly. This eases testing of the stage with simple mock functions. 

Aside of that, as you could see in diff, there was an intent to have parsers as list, and loop over specified parsers, but it seems less performant and prematurely generalized, but leaved a trail in the form of passing through the constructor.
Let's leave it untouched for now, if you don't mind?..